tsfile/src/main/java/org/apache/iotdb/tsfile/read/query/timegenerator/node/OrNode.java

```diff
@@ -19,80 +19,109 @@
 package org.apache.iotdb.tsfile.read.query.timegenerator.node;
 
 import java.io.IOException;
+import org.apache.iotdb.tsfile.common.conf.TSFileDescriptor;
+import org.apache.iotdb.tsfile.read.common.TimeColumn;
 
 public class OrNode implements Node {
 
+  private final int fetchSize = TSFileDescriptor.getInstance().getConfig()
+      .getFetchSizeOfTimeGenerator();
+
   private Node leftChild;
   private Node rightChild;
 
-  private boolean hasCachedLeftValue;
-  private long cachedLeftValue;
-  private boolean hasCachedRightValue;
-  private long cachedRightValue;
+  private TimeColumn leftTimeColumn;
+  private TimeColumn rightTimeColumn;
+
+  private TimeColumn cachedTimeColumn;
+  private boolean hasCachedValue;
+
 
   public OrNode(Node leftChild, Node rightChild) {
     this.leftChild = leftChild;
     this.rightChild = rightChild;
-    this.hasCachedLeftValue = false;
-    this.hasCachedRightValue = false;
   }
 
   @Override
-  public boolean hasNext() throws IOException {
-    if (hasCachedLeftValue || hasCachedRightValue) {
+  public boolean hasNextTimeColumn() throws IOException {
+    if (hasCachedValue) {
       return true;
     }
-    return leftChild.hasNext() || rightChild.hasNext();
-  }
 
-  private boolean hasLeftValue() throws IOException {
-    return hasCachedLeftValue || leftChild.hasNext();
+    return leftChild.hasNextTimeColumn() || rightChild.hasNextTimeColumn()
+        || leftTimeColumn.hasCurrent() || rightTimeColumn.hasCurrent();
   }
 
-  private long getLeftValue() throws IOException {
-    if (hasCachedLeftValue) {
-      hasCachedLeftValue = false;
-      return cachedLeftValue;
+  @Override
+  public TimeColumn nextTimeColumn() throws IOException {
+    if (hasCachedValue) {
+      hasCachedValue = false;
+      return cachedTimeColumn;
     }
-    return leftChild.next();
-  }
+    hasCachedValue = false;
+    cachedTimeColumn = new TimeColumn(1000);
```

Comment:
The same magic number here. You can use a constant and put it into Node.java

Reply:
done